zmon-controller-app/src/main/java/org/zalando/zmon/controller/TvTokenController.java

```diff
@@ -30,12 +33,39 @@
     private final Logger log = LoggerFactory.getLogger(TvTokenController.class);
 
     private final TvTokenService tvTokenService;
+    private final OneTimeTokenService oneTimeTokenService;
 
     private final Meter rateLimit = new Meter();
 
     @Autowired
-    public TvTokenController(TvTokenService tvTokenService) {
+    public TvTokenController(TvTokenService tvTokenService, OneTimeTokenService oneTimeTokenService) {
         this.tvTokenService = tvTokenService;
+        this.oneTimeTokenService = oneTimeTokenService;
+    }
+
+    @RequestMapping("/tv/by-email/{mail:[a-z][a-z\\.]+}")
+    public ResponseEntity<String> getByEMail(@PathVariable String mail,
+                                             @RequestHeader(name = X_FORWARDED_FOR, required = false) String bindIp,
+                                             HttpServletRequest request) {
+        if(mail == null || "".equals(mail) || mail.contains("@") || mail.contains("%40")) {
```

Comment:
Yes, true. I just did not start with the regex there that's why.

On Sunday, 3 July 2016, Henning Jacobs <notifications@github.com> wrote:

> In
> zmon-controller-app/src/main/java/org/zalando/zmon/controller/TvTokenController.java
> <https://github.com/zalando-zmon/zmon-controller/pull/211#discussion_r69394226>
> :
>
> >          this.tvTokenService = tvTokenService;
> > +        this.oneTimeTokenService = oneTimeTokenService;
> > +    }
> > +
> > +    @RequestMapping("/tv/by-email/{mail:[a-z][a-z\\.]+}")
> > +    public ResponseEntity<String> getByEMail(@PathVariable String mail,
> > +                                             @RequestHeader(name = X_FORWARDED_FOR, required = false) String bindIp,
> > +                                             HttpServletRequest request) {
> > +        if(mail == null || "".equals(mail) || mail.contains("@") || mail.contains("%40")) {
>
> unnecessary? regex will neither match empty string nor "@"
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/zalando-zmon/zmon-controller/pull/211/files/33ac2cd2f076c4fe3aa8fba74e78e0d5e321bbf3#r69394226>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABYGQwswtog1cyUAjqmqTb3t3XCGQHOcks5qR-ZogaJpZM4JD37D>
> .
>